res/games/gameitem.css

```diff
@@ -0,0 +1,138 @@
+            <img src='{{ player_country }}'>
+            </td>
+            <td align={{ alignment }} valign='middle' width='10'>
+            {% if player_global_rating is defined %}
+                ({{ player_global_rating }})
+            {% else %}
+                (???)
+            {% endif %}
+            </td>
+        {% else %}
+            <td align='left' valign='middle' width='10'>
+            {% if player_global_rating is defined %}
+                ({{ player_global_rating }})
+            {% else %}
+                (???)
+            {% endif %}
+            </td>
+            <td>
+            <img src='{{ player_country }}'>
+            </td>
+            <td align='left' width={{ width }}>
+            {{ player_login }}
+            </td>
+        {% endif %}
+         </tr>
+{% endmacro -%}
+
+
+{%- macro versus_string(fontsize) %}
+    <td align='middle' valign='middle' height='100%'>
+    <font color ='black' size={{ fontsize }}>VS</font>
+    </td>
+{% endmacro -%}
+
+
+{%- macro modtip(mods) %}
+    <b><br/>With:</b>
+    {% for mod in mods %}
+        <br/> {{ mod }}
+    {% endfor %}
+{% endmacro -%}
+
+
+{%- macro obstip(observers) %}
+    <b>Observers:</b>
+    {% for observer in observers %}
+        <br/>{{ observer.login }} <img src='{{ observer.countrypath }}'> ({{ (observer.global_rating[0]-(observer.global_rating[1]*3))|int }})
+    {% endfor %}
+{% endmacro -%}
+
+
+{%- macro formatting(teams, mods, observers) %}
+    <table border="0" cellpadding="0" cellspacing="5">
+    <tbody>
+    <tr>
+    {% if teams[1] is not defined and teams[0][2] is defined %}
+        <td valign='middle' height='100%'>
+        <font color ='black' size='+5'>FFA</font>
+        </td>
+    {% endif %}
+    <td><table>
+    {% for player in teams[0] %}
+        {{ playermacro(player,align="left") }}
+    {% endfor %}
+    </table></td>
+    {% if teams[1] %}
+        {% if teams[0][4] or teams[1][4] %}
+            {{ versus_string(fontsize='+5') }}
+	    {% elif teams[0][3] or teams[1][3] %}
+	        {{ versus_string(fontsize='+4') }}
+        {% elif teams[0][2] or teams[1][2] %}
+            {{ versus_string(fontsize='+3') }}
+	    {% elif teams[0][1] or teams[1][1] %}
+	        {{ versus_string(fontsize='+2') }}
+        {% else %}
+            {{ versus_string(fontsize='+1') }}
+        {% endif %}
+    {% endif %}
+    <td><table>
+    {% if teams[1] %}
+        {% for player in teams[1] %}
+            {{ playermacro(player,align="right") }}
+        {% endfor %}
+    {% endif %}
+    </table></td>
+    </tr>
+    {% if observers %}
+        <tr>
+        <td align='left' colspan="3">
+        {{ obstip(observers) }}
+        </td>
+        </tr>
+    {% endif %}
+    {% if mods %}
+	    <tr>
+        <td align='left' colspan="3">
+        {{ modtip(mods) }}
+        </td>
+	    </tr>
+    {% endif %}
+{% endmacro -%}
+
+
+
+{% if teams[0] is defined %}
+{{ formatting(teams, mods, observers) }}
+{% endif %}
```

Comment:
It would be nice, to have well formatted code, then it'll be much more readable.
```
{% if teams[0] is defined %}
    {{ formatting(teams, mods, observers) }}
{% endif %} 
```
or
```
{% if observers %}
    <tr>
        <td align='left' colspan="3">
            {{ obstip(observers) }}
        </td>
    </tr>
{% endif %}
```

I don't see much shame, in combining formatting html and jinja tags.

Reply:
what `teams[0]` and `teams[1]` mean? What values are inside?

Reply:
1) yes, i did that in the template with jinja syntax, but when i tried intendation with nested HTML syntax it looked garbage (to me), i guess thats personal preference

2) the teams in the data the client gets are very rudimentary packed into a list of teams.
it looks roughly like this:
teams = [[player(id,login,global,ladder),player(id,login,global,ladder)][player(id,login,global,ladder)]]
this would be a lobby where there are 2 players in team 1 (teams[0]) and 1 player in team 2 (teams[1]).